fenix/app/src/main/java/org/mozilla/fenix/search/SearchDialogController.kt

```diff
@@ -153,15 +153,24 @@ class SearchDialogController(
                     settings.shouldShowSearchShortcuts,
             ),
         )
-        fragmentStore.dispatch(
-            SearchFragmentAction.AllowSearchSuggestionsInPrivateModePrompt(
-                text.isNotEmpty() &&
-                    activity.browsingModeManager.mode.isPrivate &&
-                    settings.shouldShowSearchSuggestions &&
-                    !settings.shouldShowSearchSuggestionsInPrivate &&
-                    !settings.showSearchSuggestionsInPrivateOnboardingFinished,
-            ),
-        )
+
+        // For felt private browsing mode we're no longer going to prompt the user to enable search
+        // suggestions while using private browsing mode. The preference to enable them will still
+        // remain in settings.
+        // https://bugzilla.mozilla.org/show_bug.cgi?id=1852802
+        val isFeltPrivacyEnabled = settings.feltPrivateBrowsingEnabled
+
+        if (!isFeltPrivacyEnabled) {
```

Comment:
This got me thinking... We are putting `settings.feltPrivateBrowsingEnabled` outside of the `AllowSearchSuggestionsInPrivateModePrompt` constructor, where the rest of the settings go. Shouldn't all of the settings be outside then? Settings can't be changed between textChanges, the user has to leave the screen to change the settings. It looks like right now we are firing a lot of `action(show == false)` without ever firing `action(show == true)`. Possible follow up.

Reply:
I originally had it within but I felt that chaining 5 bools was already hard to understand and the fact that this was a feature flag is what led me to pull this one out. This setting should be getting removed in the next version after the feature rolls out.